tensorly/decomposition/tests/test_cp.py

```diff
@@ -20,13 +20,14 @@
 @pytest.mark.parametrize("init", ['svd', 'random'])
 @pytest.mark.parametrize("normalize_factors", [False, True])
 @pytest.mark.parametrize("random_state", [1, 1234])
-def test_parafac(linesearch, orthogonalise, true_rank, rank, init, normalize_factors, random_state, monkeypatch):
+@pytest.mark.parametrize("complex", [False, True])
+def test_parafac(linesearch, orthogonalise, true_rank, rank, init, normalize_factors, random_state, complex, monkeypatch):
```

Comment:
This fails under two circumstances. First, the line search routine fails to return an acceptable fit with complex values. I don't see a reason this shouldn't work, so there's possibly an error somewhere. Second, in the complex case the error is not strictly decreasing (but eventually arrives at an ok fit in the absence of line search). Perhaps the error calculation could use a second look.

Reply:
Interesting; I don't recall anyone trying out line search with complex numbers, perhaps this should be disabled by default since no research paper (as far as I know) as tackled this issue?

For the error computation, I would have to properly check but it is possible we are missing a conjugate in the mttkrp in 
https://github.com/tensorly/tensorly/blob/ce2e5e4bdb1a3d924b26c4f3cd9485e4388f2fd9/tensorly/decomposition/_cp.py#L212